Fix -Wimplicit-const-int-float-conversion error.

hsharma35 opened the pull request:
Summary: as titled

Reviewed By: skrtskrtfb

Differential Revision: D87882617

hsharma35 commented on backends/cadence/vision/kernels/kernels.cpp:
```diff
@@ -25,8 +25,8 @@ void* allocate_temp_memory(KernelRuntimeContext& ctx, size_t size) {
 T quantize(const float x, float scale, int32_t zero_point) {
-  constexpr float min_val = std::numeric_limits<T>::min();
-  constexpr float max_val = std::numeric_limits<T>::max();
+  constexpr float min_val = static_cast<float>(std::numeric_limits<T>::min());
```
This goes against executorch style guide
https://google.github.io/styleguide/cppguide.html#Type_deduction

msluszniak replied:
I will interpret this as a second point from this screen (goes from the google style guide you've sent):

![Screenshot_20251128_094820_Chrome](https://github.com/user-attachments/assets/caed2a8e-b042-4996-9e7c-6675c0b2a027)

hsharma35 replied:
I see. I'm sticking to the decision section below the text you shared about the pros/cons.

> The fundamental rule is: use type deduction only to make the code clearer or safer, and do not use it merely to avoid the inconvenience of writing an explicit type.

Saving one character (float -> auto) does not make the code more readable, IMO.

hsharma35 committed:
Fix -Wimplicit-const-int-float-conversion error. (#15982)

Summary:

as titled

Reviewed By: skrtskrtfb, zonglinpeng

Differential Revision: D87882617
